tests/Checker/Checks_Tests.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Tests for the Checks class.
+ *
+ * @package plugin-check
+ */
+
+use Exception;
+use WordPress\Plugin_Check\Checker\Checks;
+
+class Checks_Tests extends WP_UnitTestCase {
+
+	protected $checks;
+
+	public function set_up() {
+		parent::set_up();
+
+		$this->checks = new Checks( 'test-plugin/test-plugin.php' );
+	}
+
+	public function test_prepare_returns_callable() {
+		$cleanup = $this->checks->prepare();
+
+		$this->assertIsCallable( $cleanup );
+	}
+
+	public function test_get_checks_returns_array_of_expected_checks() {
+		$expected = array(
+			'example_check_class',
```

Comment:
This test assumes a class name (nit-pick: why lowercase? that's not a good convention for class names), but the code documentation is explicitly referring to class _instances_. Let's align there. I generally think instances provide more flexibility, for example since it allows to use the same class but with different configurations for a check. But there may also be some arguments for using classes (for example it would allow to only load a class PHP file if that class is actually used). Or maybe the most flexible is to allow for a mixed approach:
* Typically require a class name. In that case, the plugin checker will instantiate the class without any constructor param.
* For more flexible class usage, a closure should be provided instead which returns the necessary class instance.

We can think about which way to go with, but right now the test here is inconsistent with what is required per the documentation in the `get_checks` method. WDYT on the different approaches, which one should we go with? We should update the documentation and this test accordingly based on what we decide.